### PR TITLE
WIP: API authorization and collections creation

### PR DESCRIPTION
**Pull request by burnash**
This adds authorization to the API with help of [Django OAuth Toolkit](https://github.com/evonove/django-oauth-toolkit). The PR also implements collection creation with `POST` to `/api/collections/`.

Since CORS in NeuroVault implemented on the Nginx level, the settings differ from the ones in the Toolkit's [tutorial](https://django-oauth-toolkit.readthedocs.org/en/latest/tutorial/tutorial_01.html).

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -47,3 +47,4 @@ django-hstore==1.3.5
 cognitiveatlas
 django-datatables-view
 https://github.com/benkonrath/django-guardian/archive/django-polymorphic.zip
+django-oauth-toolkit==0.9.0
```
Comment (chrisfilo) [inline requirements.txt]: Is there any reason why the version is fixed?

**Comment by chrisfilo** [comment]
I know it's still WIP, but we'll need some documentation for app developers at some point.

**Comment by burnash** [comment]
@chrisfilo Sure, I'm writing you an email about it.

File: requirements.txt (hunk 1 above)
Comment (burnash) [inline requirements.txt]: I don't want any surprises when the newer package version is released. Do I miss some NeuroVault guidelines for dependencies?

**Comment by burnash** [comment]
@chrisfilo Looks like you've fixed the tests.

**Comment by burnash** [comment]
@chrisfilo I rebased the branch and tests are started to fail with:

~~~
FAIL: test_nidm_results_pk (neurovault.apps.statmaps.tests.test_api.Test_Atlas_APIs)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/code/neurovault/apps/statmaps/tests/test_api.py", line 217, in test_nidm_results_pk

    self.assertEqual(response['statmaps'][0][u'figure'], None)

AssertionError: u'null' != None
~~~

Before the rebase all tests passed.

**Comment by chrisfilo** [comment]
Try again I just fixed this.

**Comment by burnash** [comment]
It works! Thank you!

**Comment by burnash** [comment]
@chrisfilo I rebased the branch. I think it's ready for review.

File: requirements.txt [hunk 2]
```diff
@@ -48,3 +48,4 @@ cognitiveatlas
 django-datatables-view
 git+git://github.com/vsoch/nidmviewer.git@0.1
 https://github.com/benkonrath/django-guardian/archive/django-polymorphic.zip
+django-oauth-toolkit==0.9.0
```
Comment (chrisfilo) [inline requirements.txt]: could you add this to circleci as well? (we have this redundancy to speed up builds)

Comment (burnash) [inline requirements.txt]: Sure

**Pull request merged**